### PR TITLE
separate automatic selection of high-low side for intercomm create

### PR DESCRIPTION
**Pull request by abouteiller**
Signed-off-by: Aurelien Bouteiller <bouteill@icl.utk.edu>

### CODE REVIEW

File: ompi/communicator/comm.c [hunk 1]
```diff
@@ -1774,25 +1772,37 @@ int ompi_comm_determine_first ( ompi_communicator_t *intercomm, int high )
...
+    ompi_rte_cmp_bitmask_t mask;
+    int rc, flag;
+
+    ourproc   = ompi_group_peer_lookup(intercomm->c_local_group,0);
+    theirproc = ompi_group_peer_lookup(intercomm->c_remote_group,0);
+
+    mask = OMPI_RTE_CMP_JOBID | OMPI_RTE_CMP_VPID;
+    rc = ompi_rte_compare_name_fields(mask, (const ompi_process_name_t*)&(ourproc->super.proc_name),
+                                            (const ompi_process_name_t*)&(theirproc->super.proc_name));
+    if ( 0 > rc ) {
```
Comment (bosilca) [inline ompi/communicator/comm.c]: flag = (0 > rc) ?

Comment (abouteiller) [reply ompi/communicator/comm.c]: kept original structure of the code

**Comment by abouteiller** [comment]
This function is useful when implementing ones' own cid algorithm on an intercom. Not super useful on its own to split, but we have the need for the function in shrink, etc. and I prefer to push it upstream for conflict avoidance.

**Comment by bosilca** [comment]
My question arise from the fact that nothing in the current OMPI code or behaviors mandates this change right now. I understand that it might be used in the future, but a careful code reviewer would wonder about the justification of this particular change.

**Comment by abouteiller** [comment]
This change is part of pr #41

**Pull request closed without merging**